pkg/storaged/client.js

```diff
@@ -410,12 +410,24 @@
         }
 
         function enable_clevis_features() {
-            return cockpit.spawn([ "which", "clevis-bind-luks" ], { err: "ignore" }).then(
-                function () {
-                    client.features.clevis = true;
+            function check(help_output) {
+                client.features.clevis = true;
+                // Versions of clevis that only support LUKSv1 contain
+                // the string "Binds a LUKSv1 device" in their help
+                // output.
+                if (help_output.indexOf("Binds a LUKS device") >= 0)
+                    client.features.clevis_luksv2 = true;
+            }
+
+            // "clevis luks bind --help" writes to stderr and exits 1.
+            return cockpit.spawn([ "clevis", "luks", "bind", "--help" ], { err: "out" }).then(
```

Comment:
Eww :/ is there some plan/bug for a better interface here? (Worth linking to in a comment). This needs to force `LC_ALL=C` to avoid getting translated output.

Reply:
I suppose newer clevis with existing/old LUKSv1 installation should be no problem? That's the common case that we have to support. Could it realistically happen that someone has LUKSv2 on an older OS release? E. g. with supplying a non-default option to `cryptsetup`? I don't have a complete overview of that, so it seems to me that doing this check is safer for now (but I leave the final call to you)

Reply:
With a wrong version of clevis, Cockpit can still correctly display the key slots.  Adding a new policy might fail, but that is harmless and the error message is clear.  So let's drop the feature check.

Reply:
> the error message is clear

```
/dev/sdb (LUKS2) is not a LUKSv1 device Usage: pwmake <entropy-bits> stty: 'standard input': Inappropriate ioctl for device
```

Ouch, but I still think this is good enough.